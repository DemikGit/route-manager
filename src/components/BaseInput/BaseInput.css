.base-input__container {

}

.base-input__input {

}
