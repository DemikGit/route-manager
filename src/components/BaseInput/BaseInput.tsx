import * as React from 'react';
import { Component } from 'react';
import './BaseInput.css'


export interface IBaseInputProps {
  text: string,
  onAddPoint: () => void,
}

export class BaseInput extends Component<IBaseInputProps, object> {

  private inputRef: React.RefObject<HTMLInputElement> = React.createRef();

  public render() {
    return (
      <div className="base-input__container">
        <input
          className="base-input__input"
          ref={ this.inputRef }
          type="text"
          onKeyUp={ this.onKeyUp }
        />
      </div>
    );
  }

  private onKeyUp = ( event: React.SyntheticEvent<HTMLInputElement> ) => {
    if (event.nativeEvent instanceof KeyboardEvent) {
      if (event.nativeEvent.keyCode === 13) {
        this.props.onAddPoint();
        if (this.inputRef.current) {
          this.inputRef.current.value = '';
        }
      }
    }
  }
}
