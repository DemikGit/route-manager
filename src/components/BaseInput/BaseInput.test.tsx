import { shallow, mount } from 'enzyme';
import 'jest-enzyme';
import * as React from 'react';
import { BaseInput } from './BaseInput';

it('renders without crashing', () => {
  const onKeyUpMock = jest.fn();
  const wrapper = shallow(
    <BaseInput
      placeholder="Enter name for new routes point..."
      onKeyUp={ onKeyUpMock }
    />
  );

  expect(wrapper).toMatchSnapshot();
});

it('call onKeyUp when user types ' , () => {
  const onKeyUpMock = jest.fn().mockName('onAddPoint')

  const wrapper = shallow(
    <BaseInput
      placeholder="Enter name for new routes point..."
      onKeyUp={ onKeyUpMock }
    />
  );
  const input = wrapper.find('input');

  input.simulate('keyUp', {
    nativeEvent: {
      key: 'K',
    }
  })

  expect(onKeyUpMock).toBeCalled();
  expect(wrapper).toMatchSnapshot();
});
