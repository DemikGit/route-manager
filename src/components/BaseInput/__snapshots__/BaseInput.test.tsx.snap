// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`call onKeyUp when user types  1`] = `
<div
  className="base-input__container"
>
  <input
    className="base-input__input"
    onKeyUp={
      [MockFunction onAddPoint] {
        "calls": Array [
          Array [
            Object {
              "nativeEvent": Object {
                "key": "K",
              },
            },
          ],
        ],
      }
    }
    placeholder="Enter name for new routes point..."
    type="text"
  />
</div>
`;

exports[`renders without crashing 1`] = `
<div
  className="base-input__container"
>
  <input
    className="base-input__input"
    onKeyUp={[MockFunction]}
    placeholder="Enter name for new routes point..."
    type="text"
  />
</div>
`;
